continuous_integration/runtests.sh

```diff
@@ -1,39 +1,53 @@
-#!/usr/bin/env sh
+#!/bin/bash
 
 # License: 3-clause BSD
 
-set -e
+set -xe
+
+conda activate testenv
+
 which python
-python --version
-echo ${TOX_ENV}
+python -V
+python -c "import struct; print('platform: %d' % (8 * struct.calcsize('P')))"
+python -c "import loky; print('loky.cpu_count():', loky.cpu_count())"
+python -c "import os; print('os.cpu_count():', os.cpu_count())"
+
 
-if [ "$JOBLIB_TESTS" = "true" ]; then
+if [[ "$JOBLIB_TESTS" == "true" ]]; then
     # Install joblib from pip, patch it to use this version of loky
     # and run the joblib tests with pytest.
-    python -m venv venv/
-    source ./venv/bin/activate
-    which python
+    LOKY_PATH=$(pwd)
+
     git clone https://github.com/joblib/joblib.git src_joblib
     cd src_joblib
     pip install "pytest<7.0"  # Need to update remove occurrences of pytest.warns(None)
     pip install threadpoolctl  # required by some joblib tests
 
     pip install -e .
     export JOBLIB=`python -c "import joblib; print(joblib.__path__[0])"`
-    cp "$BUILD_SOURCESDIRECTORY"/continuous_integration/copy_loky.sh $JOBLIB/externals
-    (cd $JOBLIB/externals && bash copy_loky.sh "$BUILD_SOURCESDIRECTORY")
+    cp "$LOKY_PATH"/continuous_integration/copy_loky.sh $JOBLIB/externals
+    (cd $JOBLIB/externals && bash copy_loky.sh "$LOKY_PATH")
     pytest -vl --ignore $JOBLIB/externals --pyargs joblib
 else
     # Make sure that we have the python docker image cached locally to avoid
     # a timeout in a test that needs it.
-    if [ "$(which docker)" != "" ] && [ "$(uname)" = "Linux" ]; then
+    if [[ "$(which docker)" != "" ]] && [[ "$(uname)" = "Linux" ]]; then
```

Comment:
It's good to know you too prefer double brackets, I am happy to know I am not the only one, even if I don't really remember the reason :wink:.

Hmm actually this was a `sh` script before :scream:. Those loky people, they are so low-level :wink: